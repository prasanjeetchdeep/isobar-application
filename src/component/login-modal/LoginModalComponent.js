import React, { useState } from "react";
import { Modal, InputGroup, FormControl, Button, Alert } from "react-bootstrap";

const LoginModalComponent = ({
    show,
    onSignedIn,
    onClose
}) => {
    const [error, setError] = useState(false);
    const username = React.createRef();
    const password = React.createRef();

    const validateLogin = () => {
        if (username.current.value === "" || password.current.value === "") {
            setError(true);
        }
        else {
            setError(false);
            onSignedIn();
        }
    };

    return (
        <Modal show={show} onHide={() => onClose()}>
            <Modal.Header closeButton>
                <Modal.Title>Sign In</Modal.Title>
            </Modal.Header>
            <Modal.Body>
                {error && <Alert variant="danger">Please enter all details.</Alert>}
                <InputGroup className="mb-3">
                    <InputGroup.Prepend>
                        <InputGroup.Text id="inputGroup-sizing-default">Username</InputGroup.Text>
                    </InputGroup.Prepend>
                    <FormControl
                        ref={username}
                        aria-label="Default"
                        aria-describedby="inputGroup-sizing-default"
                    />
                </InputGroup>
                <InputGroup className="mb-3">
                    <InputGroup.Prepend>
                        <InputGroup.Text id="inputGroup-sizing-default">Password</InputGroup.Text>
                    </InputGroup.Prepend>
                    <FormControl
                        type="password"
                        ref={password}
                        aria-label="Default"
                        aria-describedby="inputGroup-sizing-default"
                    />
                </InputGroup>
                <Button variant="primary" className="float-right" onClick={() => validateLogin()}>Sign in</Button>

            </Modal.Body>

        </Modal>
    );
};

export default LoginModalComponent;
