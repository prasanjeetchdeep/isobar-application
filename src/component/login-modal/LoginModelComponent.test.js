import React from 'react';
import { render } from '@testing-library/react';
import LoginModalComponent from './LoginModalComponent';

const noop = () => { };
it("should render the login modal and show it", () => {
  const { asFragment } = render(
    <LoginModalComponent
      show={true}
      onSignedIn={noop}
      onClose={noop} />);
  expect(asFragment()).toMatchSnapshot();
});

it("should render the login modal and not show it", () => {
  const { asFragment } = render(
    <LoginModalComponent
      show={false}
      onSignedIn={noop}
      onClose={noop} />);
  expect(asFragment()).toMatchSnapshot();
});
