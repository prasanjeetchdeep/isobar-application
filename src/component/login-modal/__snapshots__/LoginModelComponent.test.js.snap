// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the login modal and not show it 1`] = `<DocumentFragment />`;

exports[`should render the login modal and show it 1`] = `<DocumentFragment />`;
